### PR TITLE
refactor(protocol-designer): add selectors for e2e tests

### PR DESCRIPTION
**Pull request by IanLondon**
# Overview / Changelog

Some new selectors for @reachojhaneha ! Serves as its own ticket.

- Add SubstepRow_aspirateWell, SubstepRow_volume, and SubstepRow_dispenseWell data-test attrs
- Add StepSelectionBannerComponent_numberStepsSelected id

# Review requests

- The 3 data-test selectors should be available on substeps
- `StepSelectionBannerComponent_numberStepsSelected` should select the element containing the "N STEPS SELECTED" in multi-select mode

# Risk assessment

Low, PD-only, just adding attrs

### CODE REVIEW

**Review by Kadee80 — APPROVED** [review]
Works locally, i think the failing tests are just macos flakiness?

File: protocol-designer/src/components/steplist/SubstepRow.js [hunk 1]
```diff
@@ -138,13 +138,19 @@ function SubstepRowComponent(props: SubstepRowProps) {
           ingreds={compactedSourcePreIngreds}
         />
 
-        <span className={styles.emphasized_cell}>
+        <span
+          className={styles.emphasized_cell}
+          data-test="SubstepRow_aspirateWell"
```
Comment (shlokamin) [inline protocol-designer/src/components/steplist/SubstepRow.js:143]: how come `Text` uses `id` but `span` uses `data-test`?

**Comment by reachojhaneha** [comment]
LGTM

**Review by reachojhaneha — APPROVED** [review]
LGTM

File: protocol-designer/src/components/steplist/SubstepRow.js (hunk 1 above)
Comment (IanLondon) [reply protocol-designer/src/components/steplist/SubstepRow.js:143]: There's only one `StepSelectionBannerComponent` instance in PD at a time so `id` is safe there, but there are almost always going to be multiple SubstepRow's so `id` wouldn't be unique, they need some other identifier that doesn't require uniqueness

Comment (shlokamin) [reply protocol-designer/src/components/steplist/SubstepRow.js:143]: makes sense!

**Review by shlokamin — APPROVED** [review]

**Pull request merged**